### DISCUSSION TITLE
Setting currentTime not working

### DISCUSSION THREAD

**Discussion by MerryMacMapface** [google/model-viewer#3551] [Q&A] [answered]
I loaded a model with an animation into the model-viewer, but am unable to set the current-time property.
When set to autoplay, I can catch the current currentTime-value, but I am not able to set it.
If not set to autoplay, the duration also is always 0, even if I manually select the correct animation-name.

Any ideas or help?
I tried it with direct setting the value and the setAttribute-method.

**Comment by elalish** [comment, marked as answer]
Duration zero sounds strange; can you post a Glitch or URL where we can repro? One thing to keep in mind is MV is a Lit web component, so it is very asynchronous. That particularly includes setting attributes - you may just need an `await mv.updateComplete;`.

**Reply by MerryMacMapface** [reply]
Oh, you're right!
An asynchronous loading works, thank you! :)

**Reply by dubi** [reply]
Hi Can you share a code example of your asynchronous loading ?
Thanks
